src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonesArr = [];
+  let stateCopy = { ...state };
+  let addCopy;
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        const addProps = obj.extraData;
+
+        addCopy = {
+          ...stateCopy,
+          ...addProps,
+        };
+
+        clonesArr.push(addCopy);
+        stateCopy = Object.assign({}, addCopy);
+
+        break;
+
+      case 'removeProperties':
+        const removeProps = obj.keysToRemove;
+
+        for (const key of Object.keys(stateCopy)) {
+          if (removeProps.indexOf(key) !== -1) {
+            delete stateCopy[key];
+          }
+        }
```

Comment:
ok, but you can do it shorter
```suggestion
        action.keysToRemove.forEach((key) => delete stateCopy[key]);
```